### PR TITLE
Add Explainer for Byte Streams

### PR DESCRIPTION
**Pull request by nidhijaju**
Add byte-streams-explainer.md.

### CODE REVIEW

**Review by ricea — COMMENTED** [review]
lgtm with one suggestion for improvement. Let's wait for @domenic to take a look.

File: byte-streams-explainer.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      controller.byobRequest.respond(v.byteLength);
+    },
+
+    autoAllocateChunkSize: DEFAULT_CHUNK_SIZE
+  });
+}
+```
+
+
+With this in hand, we can create and use BYOB readers for the returned ReadableStream. The adaptation between the
```
Comment (domenic) [inline byte-streams-explainer.md:86]:
```suggestion
With this in hand, we can create and use BYOB readers for the returned `ReadableStream`. The adaptation between the
```

File: byte-streams-explainer.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+low-level byte tracking of the underlying byte source shown here, is all taken care of automatically by the streams
```
Comment (domenic) [inline byte-streams-explainer.md:87]: This sentence doesn't seem to be entirely complete? Adaptation between what and what?

**Review by domenic — APPROVED** [review]
LGTM with nits!

File: byte-streams-explainer.md (hunk 2 above)
Comment (nidhijaju) [reply byte-streams-explainer.md]: Sorry! I just added the missing part in.

File: byte-streams-explainer.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+
+### Reading bytes from the stream into a single memory buffer
+
+The code reads the first 1024 bytes from the stream into a single memory buffer. This is due to the fact that if a
+stream is a readable byte stream, you can also acquire a BYOB reader for it, which allows more precise control over
+buffer allocation in order to avoid copies.
+
+
+```
```
[the remark this answers is not in the document]
Comment (nidhijaju) [reply byte-streams-explainer.md]: Ah okay, cool, I've added that in now :) Thank you!

**Pull request merged**